core-domain/src/main/java/org/uniprot/core/proteome/impl/ProteomeStatisticsImpl.java

```diff
@@ -0,0 +1,48 @@
+package org.uniprot.core.proteome.impl;
+
+import java.util.Objects;
+
+import org.uniprot.core.proteome.ProteomeStatistics;
+
+public class ProteomeStatisticsImpl implements ProteomeStatistics {
+    private final long reviewedProteinCount;
+    private final long unreviewedProteinCount;
+    private final long isoformProteinCount;
+
+    public ProteomeStatisticsImpl(
+            long reviewedProteinCount, long unreviewedProteinCount, long isoformProteinCount) {
```

Comment:
change the access from public to package level. This will force client to use builder class to create object of this class.